library/init/meta/interactive.lean

```diff
@@ -521,6 +517,25 @@ do x ← mk_fresh_name,
    t ← get_local x,
    induction (to_pexpr t) rec_name hs ([] : list name)
 
+private meta def cases_arg_p : parser (option name × pexpr) :=
+with_desc "(id :)? expr" $ do
+  t ← texpr,
+  match t with
+  | (local_const x _ _ _) :=
```

Comment:
Hm, I'm not sure I like that, because `have h` can always be possibly a type (before taking the actual tactic state into account), so we would have to assume this every time. This means that `let h := _` is no longer valid - it would have to be written `let h : _ := _` which I'm not so happy with. (It might be possible to eliminate the type entirely, as in `let h : := _`, but that's a bit weird to read.)